assets/scripts/fe/src/app/commons/services/payroll/payroll.service.ts

```diff
@@ -6,13 +6,15 @@ import { PAYROLL, PAYROLL_REPORT } from '../../constants/api.constants';
 import { Payroll } from '../../../commons/models/payroll.models';
 
 import { downloadFileHanlder } from '../../utils/file.utils';
+import { text } from '@angular/core/src/render3/instructions';
 
 
 @Injectable({
   providedIn: 'root'
 })
 export class PayrollService {
   public plist : any = [];
+  public mapDownloadPDF = new Map();
```

Comment:
poor selection of variable name.
- do not affix the variable by its type (only noob does that). (eg. strName, arrayList, modelMyClass)
- always think of the purpose of the variable. do not make variable for each functionality as most of the time it can be used in multiple functionalities.

`mapDownloadPDF` contains the ids of the checked items. BUT it is not being used ONLY for downloading into pdfs. it is also being used to send pdfs to the user. It's main purpose is to store selected items NOT download items. SO its better to name it `selectedPayroll`. this way it can be used to multiple functionalities.